src/Primitives.hs

```diff
@@ -401,18 +437,26 @@ primitiveMetaSet _ ctx [XObj (Sym _ _) _ _, key, _] =
 primitiveMetaSet _ ctx [target, _, _] =
   argumentErr ctx "meta-set!" "a symbol" "first" target
 
-retroactivelyRegisterInterfaceFunctions :: Context -> String -> Ty -> IO Context
-retroactivelyRegisterInterfaceFunctions ctx name t = do
+retroactivelyRegisterInterfaceFunctions :: Context -> String -> SymPath -> IO Context
+retroactivelyRegisterInterfaceFunctions ctx name interface@(SymPath _ inter) = do
   let env = contextGlobalEnv ctx
-      found = multiLookupALL name env
-      binders = map snd found
+      bindings = map snd $ Map.toList (envBindings env)
```

Comment:
Won’t this only look in the current environment, and shouldn’t it also look in all sub-environments?

Reply:
Yes! Thank you! I was misunderstanding precisely how contexts were set up so this part was confusing me greatly. Thanks for clearing it up!

Reply:
This particular case is fixed now! But I think some environment problems are still persisting when it comes to using implementation with different names than the interface. :/

Reply:
All good. I had a really dumb typo that broke nearly everything -.-